src/OutputAreaViewModel.ts

```diff
@@ -0,0 +1,374 @@
+// Copyright (c) Jupyter Development Team.
+// Distributed under the terms of the Modified BSD License.
+'use strict';
+
+import {
+  IChangedArgs, Property
+} from 'phosphor-properties';
+
+import {
+  ISignal, Signal
+} from 'phosphor-signaling';
+
+import {
+  ObservableList
+} from 'phosphor-observablelist';
+
+/**
+* A Mime bundle of data.
+*/
+export interface MimeBundle {
+  [key: string]: string;
+  'application/json'?: any;
+}
+
+/**
+* The base interface for an output view model.
+*/
+export
+class OutputBaseViewModel {
+
+  /**
+  * A signal emitted when state of the output changes.
+  */
+  stateChanged: ISignal<OutputBaseViewModel, IChangedArgs<any>>;
+
+  /**
+  * The output type.
+  */
+  outputType: string; // "execute_result" | "display_data" | "stream" | "error"
```

Comment:
You can run this as an example:
http://www.typescriptlang.org/Playground#src=%0A%0Aenum%20MyEnum%20%7B%20Foo%2C%20Bar%2C%20Baz%20%7D%0A%0Alet%20foo%20%3D%20MyEnum.Foo%3B%0A%0Aalert(%60The%20name%20is%3A%20%24%7BMyEnum%5Bfoo%5D%7D%60)%3B